src/Tools/Source/RunTests/TestRunner.cs

```diff
@@ -50,7 +49,7 @@ internal async Task<RunAllResult> RunAllAsync(IEnumerable<AssemblyInfo> assembly
             var max = (_options.TestVsi || _options.Sequential) ? 1 : (int)(Environment.ProcessorCount * 1.5);
             var cacheCount = 0;
             var waiting = new Stack<AssemblyInfo>(assemblyInfoList);
-            var running = new List<Task<TestResult>>();
+            var running = new List<(List<Process> processes, Task<TestResult> resultTask)>();
```

Comment:
I'm struggling to understand why we have the set of running `Process` entries as a sub-list here. The relationship between that list and the single `Task` for a `Process` isn't clear to me. Can you explain why having this as a separate list isn't the right approach?

Reply:
Each result is associated with 0 or more processes (0 can occur for cache cases, 2 or more can occur for VSI retries and/or procdump).

Considering this PR didn't make the build faster I'm leaning against this approach unless someone has variations to try.